torch/_inductor/codegen/triton.py

```diff
@@ -1045,21 +1045,45 @@ def mask_loads(self, mask):
         finally:
             self._load_mask = prior
 
-    def gen_assert_indirect_indexing(self, buffer, original_index, mask):
-        if mask == "None":
+    def gen_assert_indirect_indexing(self, buffer, index, mask):
+        if mask == "None" or isinstance(index, sympy.Integer):
             return
+
         body = self.current_node._body
         indirect_size = dict(zip(body.indirect_vars, body.indirect_max_sizes))
         indirect_name = body.indirect_new
+
+        var_mask = f"({mask})" if "&" in mask else mask
+
+        # Try to bound the variable via static analysis
+        # TODO We may be able to do this for dynamic shapes as well
+        if not dynamo_config.dynamic_shapes:
+            bounds = body.bounds().get_bounds()
+            indirect_bounds = {}
+            for var, rg in bounds.items():
+                var = str(var)
+                if var.startswith("set_indirect"):
+                    idx = int(var[len("set_indirect") :])
```

Comment:
what does this correspond to - `idx = int(var[len("set_indirect") :])` ?

Reply:
I'm just going from `set_indirectX` to the variable `indirectX`. This particular step is just trimming the `set_indirect` prefix.